packages/hping3/hping3.patch

```diff
@@ -0,0 +1,326 @@
+-	TCL_LIB="-ltcl${LIBPOSTFIX} -lm -lpthread"
+-fi
++#for TCLPATH_TRY in "/data/data/com.termux/files/usr/bin/"
++#do
++#	for TCLVER_TRY in "8.6" "8.3" "8.2" "8.1" "8.0"
++#	do
++	#	if [ -z $TCLSH ]
++	#	then
++		#	TCLSH_TRY=${TCLPATH_TRY}tclsh${TCLVER_TRY}
++		#	if [ -f $TCLSH_TRY ]
++			#then
++			#	TCLSH=$TCLSH_TRY
++			#	echo "===> Found Tclsh in: $TCLSH"
++			#fi
++		#fi
++	#done
++#done
++#if [ -f $TCLSH ]
++#then
++	#TCL_VER=`echo puts \\$tcl_version | $TCLSH -`
++	#USE_TCL='-DUSE_TCL'
++#	TCL_LIB="-ltcl${TCL_VER}"
++	#if [ -e $PREFIX/include/tcl${TCL_VER} ]
++	#then
++	#	TCL_INC="-I$PREFIX/include/tcl${TCL_VER}"
++	#elif [ -e $PREFIX/include/tcl.h ]
++	#then
++		#TCL_INC=""
++	#elif [ -e $PREFIX/include/tcl${TCL_VER} ]
++	#then
++		#TCL_INC="-I$PREFIX/include/tcl${TCL_VER}"
++	#else
++		#USE_TCL=""
++		#TCL_LIB=""
++		#echo "==> WARNING: no Tcl header files found!"
++	#fi
++#fi
++#if [ -n $USE_TCL ]
++#then
++	#LIBPOSTFIX=`ls -1 $PREFIX/lib/ $PREFIX/lib | grep 'libtcl[0-9]' | grep so | sed -e 's/\.so.*//g' -e 's/libtcl//g' | sort -r | head -1`
++	#TCL_LIB="-ltcl${LIBPOSTFIX} -lm -lpthread -landroid-shmem"
++#fi
+ 
+ #
+ # configurable stuff
+ #
+-PCAP="PCAP=-lpcap"
+-PCAP_INCLUDE=""
++PCAP="PCAP=-lpcap -landroid-shmem"
++PCAP_INCLUDE="-I/data/data/com.termux/files/usr/include"
+ 
+ for ARG in $*; do
+ 	case "$ARG" in
+@@ -143,6 +143,10 @@
+ 	-e "s^@TCL_INC@^$TCL_INC^g" \
+ 	-e "s^@TCL_VER@^$TCL_VER^g" \
+ 	-e "s^@TCL_LIB@^$TCL_LIB^g" \
++	-e "s^@AR@^$AR^g" \
++	-e "s^@CC@^$CC^g" \
++	-e "s^@LD@^$LD^g" \
++	-e "s^@RANLIB@^$RANLIB^g" \
+ 	<Makefile.in > Makefile
+ 
+ #
+diff -Naur hping3-20051105/libpcap_stuff.c hping3-3.0.0/libpcap_stuff.c
+--- hping3-20051105/libpcap_stuff.c	2004-04-10 05:08:56.000000000 +0530
++++ hping3-3.0.0/libpcap_stuff.c	2018-02-11 18:37:24.000000000 +0530
+@@ -17,7 +17,7 @@
+ #include <stdlib.h>
+ #include <sys/ioctl.h>
+ #include <pcap.h>
+-#include <net/bpf.h>
++#include <pcap/bpf.h>
+ 
+ #include "globals.h"
+ 
+diff -Naur hping3-20051105/scan.c hping3-3.0.0/scan.c
+--- hping3-20051105/scan.c	2003-10-22 16:11:00.000000000 +0530
++++ hping3-3.0.0/scan.c	2018-02-11 20:07:34.000000000 +0530
+@@ -20,7 +20,7 @@
+ #include <sys/ipc.h>
+ #endif
+ #include <sys/shm.h>
+-#include <sys/sem.h>
++#include <linux/sem.h>
+ #include <unistd.h>
+ #include <netdb.h>
+ #include <sys/time.h>
+@@ -34,7 +34,8 @@
+ #include <fcntl.h>
+ 
+ #if 0
+-#if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)
++#if defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED) || \
++defined(__ANDROID__)
+ /* union semun is defined by including <sys/sem.h> */
+ #else
+ /* according to X/OPEN we have to define it ourselves */
+@@ -137,7 +138,7 @@
+ {
+ 	int semid, sem_key;
+ 
+-	if ((sem_key = ftok("/tmp/hpingscansem", 1)) == -1) {
++	if ((sem_key = ftok("/data/data/com.termux/files/usr/tmp/hpingscansem", 1)) == -1) {
```

Comment:
Please replace `/data/data/com.termux/files/usr` with @TERMUX_PREFIX@.

Reply:
patch file Updated you can check :)